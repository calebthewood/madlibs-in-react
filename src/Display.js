import React, { useState } from "react";
import WordsForm from "./WordsForm";
import Madlib from "./Madlib";

function Display({madlib, update, generate}) {
  const [formShowing, setFormShowing] = useState(true);


  return (
    <div id="madlibDisplay">
      {formShowing ?
        <WordsForm
          madlib={madlib}
          update={update}
          toggleForm={()=> setFormShowing(false)}
        />
        :
        <Madlib
          title={madlib.title}
          generate={generate}
          toggleForm={()=> setFormShowing(true)}
          />}
    </div>
  );
}

export default Display;