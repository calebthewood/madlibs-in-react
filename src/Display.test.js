import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Display from "./Display";


const testMadlib = {
  "title": "Hello Test!",
  "blanks": ["noun", "adverb", "verb"],
  "value": ["Hello, ", " are you ", " operational?", 0]
}

const update = jest.fn();
const generate = jest.fn();

/** Smoke Test */
it("renders without crashing", function () {
  render(
    <Display
      madlib={testMadlib}
      update={update}
      generate={generate}
    />);
});

/** Snapshot Test */
it("snapshot", function () {
  const { container } = render(
    <Display
      madlib={testMadlib}
      update={update}
      generate={generate}
    />);

  expect(container).toMatchSnapshot();
});

it("renders", function () {
  const { container } = render(
    <Display
      madlib={testMadlib}
      update={update}
      generate={generate}
    />);
    expect(container.querySelector("#madlibDisplay")).toBeInTheDocument();
});