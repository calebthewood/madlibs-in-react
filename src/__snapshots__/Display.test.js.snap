// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    id="madlibDisplay"
  >
    <div
      class="container mb-3"
    >
      <h2>
        Hello Test!
      </h2>
      <form>
        <div>
          <div>
            <label
              class="col-form-label"
              for="0-noun"
            >
              noun
              :
            </label>
            <input
              aria-label="0-noun"
              class="form-control"
              id="0-noun"
              name="0-noun"
              placeholder="noun"
              value=""
            />
          </div>
          <div>
            <label
              class="col-form-label"
              for="1-adverb"
            >
              adverb
              :
            </label>
            <input
              aria-label="1-adverb"
              class="form-control"
              id="1-adverb"
              name="1-adverb"
              placeholder="adverb"
              value=""
            />
          </div>
          <div>
            <label
              class="col-form-label"
              for="2-verb"
            >
              verb
              :
            </label>
            <input
              aria-label="2-verb"
              class="form-control"
              id="2-verb"
              name="2-verb"
              placeholder="verb"
              value=""
            />
          </div>
        </div>
        <div
          class="d-grid gap-2 d-md-flex justify-content-md-end"
        >
          <button
            class="btn btn-primary mb-3"
            data-test-state="disabled"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
